File: frontend/src/services/enquiryService.js
import API from './api';

export const enquiryService = {
  createEnquiry: async (propertyId, message) => {
    const response = await API.post('/api/enquiries', { propertyId, message });
    return response.data;
  },

  getMyEnquiries: async () => {
    const response = await API.get('/api/enquiries/my-enquiries');
    return response.data;
  },

  getAgentEnquiries: async () => {
    const response = await API.get('/api/enquiries/agent-enquiries');
    return response.data;
  },

  respondToEnquiry: async (id, status) => {
    const response = await API.put(`/api/enquiries/${id}/respond`, null, {
      params: { status }
    });
    return response.data;
  }
};

export default enquiryService;

File: backend/src/main/java/com/realestate/service/EnquiryService.java
package com.realestate.service;

import com.realestate.dto.request.EnquiryRequest;
import com.realestate.dto.response.EnquiryResponse;
import java.util.List;

public interface EnquiryService {
    EnquiryResponse createEnquiry(EnquiryRequest request);
    EnquiryResponse respondToEnquiry(Long id, String status);
    List<EnquiryResponse> getBuyerEnquiries();
    List<EnquiryResponse> getAgentEnquiries();
}
